frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts

```diff
@@ -1,74 +1,124 @@
-import {Component, ElementRef, Input, OnDestroy, OnInit, SecurityContext, ViewChild} from "@angular/core";
-import {CalendarComponent} from 'ng-fullcalendar';
-import {Options} from 'fullcalendar';
+import {Component, ElementRef, Input, OnDestroy, OnInit, SecurityContext, ViewChild, AfterViewInit} from "@angular/core";
+import {FullCalendarComponent} from '@fullcalendar/angular';
 import {States} from "core-components/states.service";
 import {IsolatedQuerySpace} from "core-app/modules/work_packages/query-space/isolated-query-space";
 import {untilComponentDestroyed} from "ng2-rx-componentdestroyed";
 import {WorkPackageResource} from "core-app/modules/hal/resources/work-package-resource";
 import {WorkPackageCollectionResource} from "core-app/modules/hal/resources/wp-collection-resource";
 import {WorkPackageViewFiltersService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-filters.service";
 import * as moment from "moment";
-import {Moment} from "moment";
 import {WorkPackagesListService} from "core-components/wp-list/wp-list.service";
 import {StateService} from "@uirouter/core";
-import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {NotificationsService} from "core-app/modules/common/notifications/notifications.service";
 import {DomSanitizer} from "@angular/platform-browser";
 import {WorkPackagesListChecksumService} from "core-components/wp-list/wp-list-checksum.service";
 import {OpTitleService} from "core-components/html/op-title.service";
+import dayGridPlugin from '@fullcalendar/daygrid';
+import { EventInput, EventApi } from '@fullcalendar/core';
+import { EventSourceError } from '@fullcalendar/core/structs/event-source';
+import { take } from 'rxjs/operators';
+import { ToolbarInput } from '@fullcalendar/core/types/input-types';
+import {ConfigurationService} from "core-app/modules/common/config/configuration.service";
+
+interface CalendarViewEvent {
+  el:HTMLElement;
+  event:EventApi;
+  jsEvent:MouseEvent;
+}
 
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/65420bc156cbc4b6a13f40aa608d3e38780d29d5/frontend/src/app/modules/calendar/wp-calendar/wp-calendar.component.ts#L29"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>